sysfs/class_power_supply_test.go

```diff
@@ -72,9 +72,7 @@ func TestPowerSupplyClass(t *testing.T) {
 		},
 	}
 
-	if !reflect.DeepEqual(powerSupplyClass, psc) {
-		want, _ := json.Marshal(powerSupplyClass)
-		get, _ := json.Marshal(psc)
-		t.Errorf("Result not correct: want %v, have %v.", string(want), string(get))
+	if diff := cmp.Diff(want, got); diff != "" {
```

Comment:
By the way, I introduced this because it produces super helpful output for tests, such as this one (with temporarily modified code):

```
=== RUN   TestPowerSupplyClass
--- FAIL: TestPowerSupplyClass (0.00s)
    class_power_supply_test.go:76: unexpected power supply class (-want +got):
          sysfs.PowerSupplyClass{
                "AC": {Name: "AC", Online: &0, Type: "Mains"},
                "BAT0": {
                        ... // 39 identical fields
                        PowerNow:         &4830000,
                        PrechargeCurrent: nil,
        -               Present:          &2,
        +               Present:          &1,
                        Scope:            "",
                        SerialNumber:     "38109",
                        ... // 24 identical fields
                },
          }
```